### ISSUE TITLE
Confusing: RBMessageNode>>#superOf

### ISSUE THREAD

**Issue by MarcusDenker** [guillep/pharo-core#14213]
**Describe the problem**
RBMessageNode>>#superOf is quite confusing. This is the class where the method is installed in ("methodNode methodClass")

OCASTTranslator>>#emitMessageNode: reads now

```
aMessageNode isSuperSend
		ifTrue: [methodBuilder send: aMessageNode selector toSuperOf: aMessageNode superOf ]
		ifFalse: [methodBuilder send: aMessageNode selector]
```

"toSuperOf: aMessageNode superOf" seems wrong.

**TODO**
- The name #superOf is bad on the level of RBMessageNode. It should be #methodClass
- Do we need to cache it? the ivar is set in OCASTSemanticAnalyzer>>#visitMessageNode:, but we can just implement #methodClass as "^self methodNode methodClass" 

The code then looks like:

```
aMessageNode isSuperSend
		ifTrue: [methodBuilder send: aMessageNode selector toSuperOf: aMessageNode methodClass ]
		ifFalse: [methodBuilder send: aMessageNode selector]
```

**Comment by privat** [comment]
agree. the name superOf is superBad.

The idea of attaching the class in the node was to make it more independent of its context, like detaching/attaching AST nodes.
I think for instance of some compilation-level method inlining. If you statically inline a method that does a supercall you need to remember the original method class somewhere (I do not even know if the bytecode allow having super call not on self).
But maybe it's just fragile and too weird

**Comment by MarcusDenker** [comment]
yes, with the class in the MessageNode, one can use (or ab-use?) this to do e.g. sends that skip more super classes. Sadly a real static send is not possible as we can not use another receiver than "super", but maybe we can play a bit... creating an anonymous subclass on the fly is possible, if we change #isSuperSend we might be able to do real static sends... ah, but only of we force the byte-code that takes the class from the stack 

Maybe we should explore if it is possible to always use that byte-code... it is used now as soon as super is used in a non-optimized block, as the blocks do not have the class in the last literal. This would give us more freedom to use the last literal always for additional state, for example (one less thing where the VM relies on the structure of compiledMethod).

Maybe we should fix for now just the name of the accessor (and the ivar name) to fix this issue of understandability, and let for now the ivar.
